packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -941,7 +951,10 @@ class CachedLocalEngineArtifacts implements LocalEngineArtifacts {
   }
 
   String _genSnapshotPath() {
-    const List<String> clangDirs = <String>['.', 'clang_x64', 'clang_x86', 'clang_i386', 'clang_arm64'];
+    // Does this  new implementation break anything ?
+    // Can we work TargetPlatform and EnvironmentType into this conditionally somehow ?
+    // Will the follwoing break anything: return the gen_snapshot inside clang_x64 directory if targetPlatform == linux_arm64
+    const List<String> clangDirs = <String>['clang_x64', 'clang_x86', 'clang_i386', 'clang_arm64','.'];
```

Comment:
why did you move `.` to the end here?

Reply:
Because I don't want to grab the `gen_snapshot` in the root engine build output folder , but rather the one in the 'clang_arm64' directory.

Reply:
Sorry I'm missing some context, but I think you'd get a `gen_snapshot` under a `clang_arm64` directory in the build output if you're building on an `arm64` host. Is that the situation? That `gen_snapshot` should be targeting `arm64` but you can check by running `gen_snapshot --version`. It will show the target configuration in quotes at the end of the version string.

If you don't want the `gen_snapshot` under `.`, I'm curious what it is. Would you mind giving the output of `file gen_snapshot` and `gen_snapshot --version` for the `gen_snapshot` under `.`?

Reply:
### For  `.`

**command:**
`/out/linux_debug_arm64$ file gen_snapshot`
 **output:**
`gen_snapshot: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, BuildID[sha1]=c3db3f84ed98ba1c97b8af997bdfb2acb5231e7a, stripped
`

**command:**
`linux_debug_arm64$ ./gen_snapshot --version`
 **output:**
`bash: ./gen_snapshot: cannot execute binary file: Exec format error`

Do you need me to run this on arm64 to get the version , or is this enough info ?

### For `clang_x64`

**command:**
`out/linux_debug_arm64/clang_x64$ file gen_snapshot`
 **output:**
`gen_snapshot: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=04d12c6e9a3200644565d5c91cbd331cb8bc9a55, stripped
`

**command:**
`linux_debug_arm64/clang_x64$ ./gen_snapshot --version`
 **output:**
`Dart SDK version: 2.16.0-46.0.dev (dev) (Wed Nov 24 07:17:06 2021 -0800) on "linux_simarm64"
`